6211 Маркова Юлия Сергеевна Лаб.4 Вар.16

marulias opened the pull request:
ФИО: Маркова Юлия Сергеевна
Номер группы: 6211
Номер лабораторной работы: 4
Номер варианта: 16
Краткое описание решаемой задачи: Согласно варианту 16 лабораторной работы № 4, мне нужно было добавить столбец со значениями для гистограммы распределения соотношения сторон изображения (отношение длины к ширине или ширины к длине).
Краткое описание работы кода: Код создает таблицу с путями к изображениям, вычисляет соотношения сторон и распределяет их по диапазонам. Функции сортировки и фильтрации позволяют упорядочить данные по соотношению сторон и отфильтровать по нужному диапазону. Затем строится гистограмма распределения и результаты сохраняются в файлы CSV и PNG.

marulias committed:
lab1

marulias committed:
Update lab1.py

marulias committed:
Update lab1.py

marulias committed:
Update lab1.py

исправила код по вашим замечаниям

marulias committed:
lab2

marulias committed:
Delete lab2 directory

marulias committed:
lab4

microalan reviewed (commented):
Все те же замечания, что и к предыдущим лабам + диапазоны задаёт пользователь и данные считывать из файла-аннотации

P.s. а что с кодом? Почему он на русском?

marulias committed:
Delete lab1 directory

marulias committed:
Delete lab4 directory

marulias committed:
lab4

marulias committed:
Update functions.py

разбила на модули, добавила исключения, теперь пользователь задает диапазон и данные считываем из файла аннотации

microalan commented on lab4/functions.py:
```diff
@@ -0,0 +1,53 @@
+                    width, height = img.size
+                    aspect_ratios.append(round(width / height, 2))
+            except Exception as e:
+                print(f"Ошибка при обработке изображения {path}: {e}")
```
Убрать print

microalan commented on lab4/main.py:
```diff
@@ -0,0 +1,60 @@
+
+        # Пользователь вводит диапазоны
+        print("\nВведите границы диапазонов через пробел (например: 0.5 1.0 1.5 2.0 2.5):")
+        user_input = input().split()
```
Передавать диапазон через аргументы командной строки

microalan commented on lab4/main.py:
```diff
@@ -0,0 +1,60 @@
+
+        # Сохранение и отображение графика
+        plt.savefig('гистограмма.png')
+        plt.show()
```
Выделить в функцию

microalan reviewed (commented):
1. Добавить docstring
2. Переименовать колонки (правила именования такие же как для переменных в python)

marulias committed:
Update functions.py

исправила замечания

marulias committed:
Update main.py

microalan reviewed (commented):
Добавить docstring

marulias committed:
Update functions.py

добавила докстринг

marulias committed:
Update main.py

microalan reviewed (approved):
Зачтено
